# Worksheet #5

More Operations on Arrays


In [1]:
# the usual necessities

%matplotlib inline
import numpy as np
import pylab

Each value in the array below (pictures) represents the number of pictures an individual who went on a three week trip to Japan took during the duration of that trip.

In [2]:
pictures = np.array([233, 689, 382, 561,  65, 920, 318, 596, 256, 500, 638, 881, 753,
       902, 256, 857,  32, 645, 348, 994, 183, 728, 507, 231, 238, 581,
       700, 762, 984, 428, 253, 329, 417, 662, 994, 450,  34, 657, 202,
       805, 732, 744, 856, 758, 664, 670, 570, 444, 489, 451, 369, 461,
       685, 494, 332, 939, 843, 410, 749, 949,  29, 145, 561, 783, 309,
       406, 435,  63,  50, 936, 779, 426, 110, 155, 211, 764,  53,  35,
       207, 798, 214, 501, 513, 773, 178, 659, 186, 274, 721, 731, 757,
        94, 847, 155, 510, 763, 243, 430, 406, 220, 757, 338, 410, 537,
       553, 973, 446, 436, 220, 231, 835, 814, 865, 965, 978,  17, 324,
       406, 264, 532, 835, 438,   5, 271, 852,  61, 366, 954, 719, 839,
       832,  50, 459, 277, 940, 962, 196, 153, 881, 179, 568, 919, 496,
       928, 954, 923, 992, 519, 465, 784, 113,  19, 505, 842, 299,  38,
       971, 521, 169, 563, 373, 278, 786, 510, 433, 844, 303, 327, 116,
       684, 356, 608, 185, 763, 447, 329, 291, 825,  18,  94, 831, 780,
       661, 628, 751, 960, 628,  40, 122, 962, 418, 340, 455, 578, 807,
       451, 544, 636, 124, 434, 450,  24, 787, 983, 742, 501, 565, 419,
       665, 454, 110, 202, 639, 869, 873, 659, 892, 149, 640, 429, 664,
       125, 884, 408, 994,  72, 575, 546, 702, 338, 759, 177, 930, 452,
       357, 610,  52, 848, 122, 900])
pictures

array([233, 689, 382, 561,  65, 920, 318, 596, 256, 500, 638, 881, 753,
       902, 256, 857,  32, 645, 348, 994, 183, 728, 507, 231, 238, 581,
       700, 762, 984, 428, 253, 329, 417, 662, 994, 450,  34, 657, 202,
       805, 732, 744, 856, 758, 664, 670, 570, 444, 489, 451, 369, 461,
       685, 494, 332, 939, 843, 410, 749, 949,  29, 145, 561, 783, 309,
       406, 435,  63,  50, 936, 779, 426, 110, 155, 211, 764,  53,  35,
       207, 798, 214, 501, 513, 773, 178, 659, 186, 274, 721, 731, 757,
        94, 847, 155, 510, 763, 243, 430, 406, 220, 757, 338, 410, 537,
       553, 973, 446, 436, 220, 231, 835, 814, 865, 965, 978,  17, 324,
       406, 264, 532, 835, 438,   5, 271, 852,  61, 366, 954, 719, 839,
       832,  50, 459, 277, 940, 962, 196, 153, 881, 179, 568, 919, 496,
       928, 954, 923, 992, 519, 465, 784, 113,  19, 505, 842, 299,  38,
       971, 521, 169, 563, 373, 278, 786, 510, 433, 844, 303, 327, 116,
       684, 356, 608, 185, 763, 447, 329, 291, 825,  18,  94, 83

# 1. How many people took more than 300 photos during their trip?

In [4]:
np.count_nonzero(pictures > 300)

176

I got 176.

# 2. How many people took less photos than the average?

In [5]:
AveragePictures = np.mean(pictures)
np.count_nonzero(pictures < AveragePictures)

124

I got 124.

# 3. In actuality, each set of 20 people was in a tour bus group during the trip.  Convert this array into a 2D array where each row is the tour group that person was in and each column is the seat that person sat in on the tour bus. To check, how many tour groups were there?

In [18]:
TourGroups= np.reshape(pictures,(12,20))
TourGroups.shape

(12, 20)

I got 12 tour groups. Make sure you actually change the shape of your array to be two dimensional. (12,20)

# 4. What was the greatest number of photos taken by any one group?

In [19]:
np.max(np.sum(TourGroups, axis=1))

12609

I got 12609.

# 5. What was the average number of photos taken by people sitting in the third seat of the tour bus?

In [20]:
np.mean(TourGroups[:,2])

570.66666666666663

I got about 571. Hint: Remember that python arrays start indexing at 0 and not 1!

# 6. How many groups took more than 10,000 photos?

In [21]:
LotsOfPictures = np.sum(TourGroups, axis=1)> 10000
len(TourGroups[LotsOfPictures])

9

I got 9 groups.

Submit your notebook below

In [22]:
import comp116
comp116.submit('ws5')